core/src/main/scala/org/bitcoins/core/protocol/ln/LnHumanReadablePart.scala

```diff
@@ -0,0 +1,84 @@
+package org.bitcoins.core.protocol.ln
+
+import org.bitcoins.core.config.NetworkParameters
+import org.bitcoins.core.protocol.NetworkElement
+import org.bitcoins.core.protocol.ln.LnParams._
+import scodec.bits.ByteVector
+import scala.util.matching.Regex
+import scala.util.{ Failure, Try }
+
+sealed abstract class LnHumanReadablePart extends NetworkElement {
+  def network: LnParams
+
+  def amount: Option[LnCurrencyUnit]
+
+  override def toString: String = network.value + invoiceAmount
+
+  def invoiceAmount: String = {
+    if (amount.isDefined) {
+      amount.get.toBigInt.toString() + amount.get.character
```

Comment:
Makes total sense. Do you want that functionality as part of this PR, or separately?

Reply:
I already forgot, we actually have a `.toEncodedString` defined in `LnCurrencyUnit`, which does essentially this. However, it does currently return a 0 value, which I dont think would be desirable in this case. We would rather just return a `""` in that case.

Reply:
I'm not sure if I understand the case where it is valid to return just a `""`. I think we should probably encode these methods as `Option[String]` instead and return `None` instead of `""` in the case that we don't have anything too return. In general, we try to encode things at the type level using `Option` rather than doing it at runtime with a `null`, `""`, `-1` or whatever your preferred C idiom is :-)